From d90594b89df6654884890652934f44e1074a964b Mon Sep 17 00:00:00 2001
From: Eugeniy Paltsev <Eugeniy.Paltsev@synopsys.com>
Date: Mon, 27 Jan 2020 20:05:26 +0300
Subject: [PATCH 20/23] HDMI: allow pixel clocks != 27MHz

Signed-off-by: Eugeniy Paltsev <Eugeniy.Paltsev@synopsys.com>
---
 drivers/gpu/drm/dw_hdmi.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/drivers/gpu/drm/dw_hdmi.c b/drivers/gpu/drm/dw_hdmi.c
index 3ca7a1ac8b1e..62505a0a1e17 100644
--- a/drivers/gpu/drm/dw_hdmi.c
+++ b/drivers/gpu/drm/dw_hdmi.c
@@ -70,16 +70,9 @@ static const struct dw_hdmi_phy_config snps_phy_config[] = {
 static enum drm_mode_status dw_hdmi_mode_valid(struct drm_connector *con,
 					       const struct drm_display_mode *mode)
 {
-	const int delta = 500;
-
 	dev_info(con->dev->dev, "PAL: %s clock is: %d KHz\n",
 		 __func__, mode->clock);
 
-	if (mode->clock < (27000 - delta))
-		return MODE_CLOCK_LOW;
-	if (mode->clock > (27000 + delta))
-		return MODE_CLOCK_HIGH;
-
 	return MODE_OK;
 }
 
-- 
2.16.2

